test/distributed/org/apache/cassandra/distributed/impl/InstanceMetrics.java

```diff
@@ -45,16 +45,32 @@
         this.metricsRegistry = metricsRegistry;
     }
 
+    @Override
     public List<String> getNames()
     {
         return new ArrayList<>(metricsRegistry.getNames());
     }
 
+    @Override
     public long getCounter(String name)
     {
-        return metricsRegistry.getCounters().get(name).getCount();
+        Counter counter = metricsRegistry.getCounters().get(name);
+        if (counter != null)
+            return counter.getCount();
+        Meter meter = metricsRegistry.getMeters().get(name);
+        if (meter != null)
+            return meter.getCount();
+        Histogram histogram = metricsRegistry.getHistograms().get(name);
+        if (histogram != null)
+            return histogram.getCount();
+        Timer timer = metricsRegistry.getTimers().get(name);
+        if (timer != null)
+            return timer.getCount();
+        //TODO no way to handle not found other than NPE?
```

Comment:
TODO left over - what to do?

Reply:
this was taken out of https://issues.apache.org/jira/browse/CASSANDRA-16850 and asked @ifesdjeen to review it there

Reply:
this is mostly me documenting that the jvm-dtest api exposes a `long` so "does not exist" returns a NPE atm (and without this patch)